inference-engine/src/mkldnn_plugin/nodes/mkldnn_roi_pooling_node.cpp

```diff
@@ -677,13 +634,88 @@ void MKLDNNROIPoolingNode::execute() {
                 }
             }
 
-            if (roi_pooling_kernel) {
-                (*roi_pooling_kernel)(&arg);
+            if (execPtr) {
+                execPtr->exec(arg);
             }
         }
     });
 }
 
+void MKLDNNROIPoolingNode::executeDynamicImpl(mkldnn::stream strm) {
+    return execute(strm);
+}
+
+void MKLDNNROIPoolingNode::prepareParams() {
+    auto& srcMemPtr0 = getParentEdgeAt(0)->getMemoryPtr();
+    auto& srcMemPtr1 = getParentEdgeAt(0)->getMemoryPtr();
+    auto& dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+    if (!srcMemPtr0 || !srcMemPtr0->GetPrimitivePtr())
+        IE_THROW() << "Input memory didn't allocate.";
+    if (!srcMemPtr1 || !srcMemPtr1->GetPrimitivePtr())
+        IE_THROW() << "Input memory didn't allocate.";
+    if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
+        IE_THROW() << "Destination memory didn't allocate.";
+    if (getSelectedPrimitiveDescriptor() == nullptr)
+        IE_THROW() << "Preferable primitive descriptor is not set.";
+
+    auto selectedPD = getSelectedPrimitiveDescriptor();
+    if (!selectedPD)
+        IE_THROW() << "CPU ROI Pooling node with name '" << getName() << "' doesn't have primitive descriptors.";
+
+    refParams.c_block = mayiuse(cpu::x64::avx512_common) ? 16 : 8;;
+    refParams.nb_c_blocking = mayiuse(cpu::x64::avx512_common) ? 15 : 7;
+    refParams.alg = getAlgorithm();
+
+    const auto& config = selectedPD->getConfig();
+    refParams.src_prc = config.inConfs[0].desc->getPrecision();
+    refParams.dst_prc = config.outConfs[0].desc->getPrecision();
+    refParams.src_data_size = refParams.src_prc.size();
+    refParams.dst_data_size = refParams.dst_prc.size();
```

Comment:
since we decided to leave `jit_roi_pooling_params` as class field I guess we can fill this fields inside `createPrimitive`

Reply:
Done, thanks!